File: elixir/words/lib/numerical.ex
defmodule Numerical do
  @moduledoc """
  Documentation for `Numerical`.
  """
  def fibonacci(n) when n < 0, do: -1
  def fibonacci(n) when n < 2, do: n
  def fibonacci(n), do: fibonacci_calc(n, 0, 1)

  defp fibonacci_calc(2, acc2, acc1), do: acc1 + acc2
  defp fibonacci_calc(n, acc2, acc1), do: fibonacci_calc(n - 1, acc1, acc1 + acc2)

  def factorial(n) when n < 0, do: -1
  def factorial(0), do: 1
  def factorial(n), do: factorial_calc(n, 1)

  defp factorial_calc(1, acc), do: acc
  defp factorial_calc(n, acc), do: factorial_calc(n - 1, n * acc)

  def gcd(a, 0), do: abs(a)
  def gcd(a, b), do: gcd(b, rem(a, b))

  def lcm(a, b), do: div(a, gcd(a, b)) * b
end

File: c/words/src/numerical.c
#include "../include/numerical.h"

int fibonacci (int n)
{
    if (n < 0)
    {
        return -1;
    } else if (n < 2)
    {
        return n;
    }
    int acc2 = 0;
    int acc1 = 1;
    int aux = 0;
    while (n > 2)
    {
        aux = acc1 + acc2;
        acc2 = acc1;
        acc1 = aux;
        n--;
    }
    return acc1 + acc2;
}

int factorial (int n)
{
    if (n < 0)
    {
        return -1;
    }
    int acc = 1;
    while (n > 1)
    {
        acc = n * acc;
        n--;
    }
    return acc;
}

int gcd (int a, int b)
{
    int aux = 0;
    while (b != 0)
    {
        aux = a % b;
        a = b;
        b = aux;
    }
    return a;
}

int lcm (int a, int b)
{
    return (a / gcd(a, b)) * b;
}